### PR TITLE
Implement backend routes to CRUD questions

### PR DESCRIPTION
**Pull request by asaierika**
Implement routes in`/src/routes/questions.ts`.

### CODE REVIEW

**Comment by asaierika** [comment]
Would it be better to put it in a file in the `src` folder instead of the `src/routes` subfolder?

File: services/question/src/routes/questions.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+router.post('/', async (req, res) => {
+  try {
+    const questionData: Question = req.body
+    const docRef = await db.add(questionData)
+    res.status(201).json({ id: docRef.id, ...questionData })
+  } catch (error) {
+    res.status(500).json({ error: 'Failed to create question' })
+  }
+})
```
Comment (bachletuankhai) [inline services/question/src/routes/questions.ts:16]: Should we protect this route? Not everyone should be able to add new questions.

Comment (bachletuankhai) [reply services/question/src/routes/questions.ts:16]: Same thing to `put` and `delete`.

**Comment by bachletuankhai** [comment]
> Would it be better to put it in a file in the `src` folder instead of the `src/routes` subfolder?

What is "it" you are referring to here?

File: services/question/src/routes/questions.ts (hunk 1 above)
Comment (bachletuankhai) [reply services/question/src/routes/questions.ts:16]: I think we can use some sort of user token in the header to resolve this. Maybe we can check the provided user service on how they implement token-based authorization and apply it to private routes here.

Comment (asaierika) [reply services/question/src/routes/questions.ts:16]: Sure! I'll implement this.

**Comment by asaierika** [comment]
> > Would it be better to put it in a file in the `src` folder instead of the `src/routes` subfolder?
> 
> What is "it" you are referring to here?

I'm referring to the file `/src/routes/questions.ts`

**Comment by bachletuankhai** [comment]
> > > Would it be better to put it in a file in the `src` folder instead of the `src/routes` subfolder?
> 
> > 
> 
> > What is "it" you are referring to here?
> 
> 
> 
> I'm referring to the file `/src/routes/questions.ts` 

It's okay to have that `routes` folder. And with public and private routes, I think you even need to split the routes into 2 different routers and have 2 more subdirectory here.

File: services/question/src/routes/questions.ts (hunk 1 above)
Comment (ys112) [reply services/question/src/routes/questions.ts:16]: > I think we can use some sort of user token in the header to resolve this. Maybe we can check the provided user service on how they implement token-based authorization and apply it to private routes here.

if protecting route involves user service, I think this fall under D3 milestone?

Comment (asaierika) [reply services/question/src/routes/questions.ts:16]: > > I think we can use some sort of user token in the header to resolve this. Maybe we can check the provided user service on how they implement token-based authorization and apply it to private routes here.
> 
> if protecting route involves user service, I think this fall under D3 milestone?

I think maybe we can implement this for the question service routes first for D2.

Comment (bachletuankhai) [reply services/question/src/routes/questions.ts:16]: No user service involved in D2, so this will be moved to D3

**Review by ys112 — APPROVED** [review]

**Review by bachletuankhai — APPROVED** [review]

**Pull request merged**